thorlcr/activities/indexwrite/thindexwriteslave.cpp

```diff
@@ -452,6 +466,13 @@ class IndexWriteSlaveActivity : public ProcessSlaveActivity, public ILookAheadSt
 
                     CMessageBuffer msg;
                     MemoryAttr dummyRow;
+                    size32_t keyedSize = helper->getKeyedSize();
+                    if (keyedSize == (unsigned) -1)
+                    {
+                        // For some reason, if there is no payload, getKeyedSize returns -1
+                        const RtlRecord &indexRecord = helper->queryDiskRecordSize()->queryRecordAccessor(true);
+                        keyedSize = indexRecord.getFixedOffset(indexRecord.getNumKeyedFields());
+                    }
```

Comment:
perhaps this block should move to start() as same as block in open(), and record keyedSize as member and use here and in open()